gpytorch/settings.py

```diff
@@ -681,6 +681,17 @@ class skip_logdet_forward(_feature_flag):
     _default = False
 
 
+class symeig_dtype(_value_context):
```

Comment:
Seems like this could lead to an inflation of value contexts if we do this for every operation. Would it be possible to group these together (similar to `fast_computations`?)

Reply:
Yes, ultimately, but from what I see now, it's likely the only value context that does computations in double. I could be wrong though.

Reply:
Hmm.. it does look like some choleskies are stored / performed in double, particularly these https://github.com/cornellius-gp/gpytorch/blob/4685d383f5cb28a39c8ef2b564881e2f0aa32aa7/gpytorch/variational/variational_strategy.py#L71 .

Reply:
If it's not too much work I'd advocate for generating an umbrella `settings.dtypes` registry so we can do things like `settings.dtypes.cholesky` or `settings.dtypes.symeig` from the start rather than introducing this and then having to deprecate it. But it that's annoying/difficult we can also merge this in as is.